### PR TITLE
Add metrics around batch creation and eventually received replies [cherry-pick to previewnet]

### PR DESCRIPTION
**Pull request by igor-aptos**
shared cherry-pick first, as PR on main (#7091) will need to rebase on top of @zekun000's changes

### CODE REVIEW

File: consensus/src/quorum_store/proof_coordinator.rs [hunk 1]
```diff
@@ -38,21 +38,24 @@ pub(crate) type ProofReturnChannel = oneshot::Sender<Result<(ProofOfStore, Batch
 struct IncrementalProofState {
     info: SignedDigestInfo,
     aggregated_signature: BTreeMap<PeerId, bls12381::Signature>,
+    aggregated_voting_power: u128,
```
Comment (zekun000) [inline consensus/src/quorum_store/proof_coordinator.rs:41]: do we need to cache this? checking voting power along should be very fast, it's just sum over hashmap lookups

Comment (igor-aptos) [reply consensus/src/quorum_store/proof_coordinator.rs:41]: @zekun000 - I haven't measured - but:
* we are checking voting power much more often.
* it is an O(n^2) operation currently. going through a hashmap with 70 elements is not trivially cheap, might compare or be more expensive than signature verification, etc.

Comment (zekun000) [reply consensus/src/quorum_store/proof_coordinator.rs:41]: I think it's fine to cache it, but going through a hashmap with 70 elements should be magnitudes cheaper than a signature verification

**Review by zekun000 — APPROVED** [review]

File: consensus/src/quorum_store/proof_coordinator.rs [hunk 2]
```diff
@@ -64,40 +71,78 @@ impl IncrementalProofState {
...
 
     fn ready(&self, validator_verifier: &ValidatorVerifier, my_peer_id: PeerId) -> bool {
-        self.aggregated_signature.contains_key(&my_peer_id)
-            && validator_verifier
-                .check_voting_power(self.aggregated_signature.keys())
-                .is_ok()
+        if self.aggregated_signature.contains_key(&my_peer_id)
+            && self.aggregated_voting_power >= validator_verifier.quorum_voting_power()
+        {
+            let recheck = validator_verifier.check_voting_power(self.aggregated_signature.keys());
```
[the remark this answers is not in the document]
Comment (igor-aptos) [reply consensus/src/quorum_store/proof_coordinator.rs:106]: where exactly?  If I am not unwrapping , and only having one side - using if recheck.is_err() seems cleanest?

Comment (sitalkedia) [inline consensus/src/quorum_store/proof_coordinator.rs:106]: Can you clarify why the recheck is needed?When is the  discrepancy expected?

Comment (igor-aptos) [reply consensus/src/quorum_store/proof_coordinator.rs:106]: never. that's why I have error!() log.

[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/src/quorum_store/proof_coordinator.rs:106]:
```
match validator_verifier.check_voting_power(self.aggregated_signature.keys()) {
  Ok(()) => true,
  Err(e) => {
    error!(...);
    false
  }
}
```

Comment (sitalkedia) [reply consensus/src/quorum_store/proof_coordinator.rs:106]: Hmm should we panic! instead in that case?

**Pull request merged**